### ISSUE TITLE
Working with collections

### ISSUE THREAD

**Issue by julillosamaral** [cloudcreativity/json-api#21]
Hi,
I created a client which is receiving a response similar to this: 

``` json
 {
    "data": [
        {
            "type": "ENTITY",
            "id": "identifier",
            "attributes": {
                "name": "My name 1",
                "number": 1,
            },
            "relationships": {
                "relation": {
                    "data": {
                        "type": "type",
                        "id": "identifier2"
                    }
                },
            },
            "links": {
                "self": "link1"
            }
        },
        {
            "type": "ENTITY",
            "id": "identifier2",
            "attributes": {
                "name": "My name 2",
                "number": 2,
            },
            "relationships": {
                "relation": {
                    "data": {
                        "type": "type",
                        "id": "identifier2"
                    }
                },
            },
            "links": {
                "self": "link1"
            }
        },
        {
            "type": "ENTITY",
            "id": "identifier",
            "attributes": {
                "name": "My name 3",
                "number": 3,
            },
            "relationships": {
                "relation": {
                    "data": {
                        "type": "type",
                        "id": "identifier3"
                    }
                },
            },
            "links": {
                "self": "link1"
            }
        },        
    ]
}
```

As you can see, inside data I receive an array of Entities, I don't know how I can iterate in this entities I receive in the array.
It would be great to have an example on how to do that!

Thanks in advance,
Julio

**Comment by lindyhopchris** [comment]
Hi!

I'm assuming you're on the latest version of the library (if not, just say). To get the array from the `data` key, you can do:

``` php
use CloudCreativity\JsonApi\Decoders\DocumentDecoder;
use CloudCreativity\JsonApi\Object\Resource;

/** @var string $content    the response content */
$decoder = new DocumentDecoder();
$document = $decoder->decode($content);

/** @var stdClass[] $data */
$data = $document->get('data');

/** If you want these as actual resource objects... */
$resources = array_map(function (stdClass $object) {
    return new Resource($object);
}, $data);
```

The reason there isn't an inbuilt way of doing this is because the library has been built to validate content coming _from_ a client, rather than a client handling the response from a server. However, this is definitely an intended use case for the package, we just haven't put a huge amount of dev time into it yet. 

It'd be a great addition if I added support for this in, because what you're trying to do is an obvious omission for the use case of handling responses

All I need to do is add a collection that holds instances of this class:
https://github.com/cloudcreativity/json-api/blob/master/src/Contracts/Object/ResourceInterface.php

And then add a method to this interface:
https://github.com/cloudcreativity/json-api/blob/master/src/Contracts/Object/DocumentInterface.php
To get the collection from the `data` key.

I'm happy to do this but I won't be able to get to it until the weekend at the earliest, or next week.

**Comment by lindyhopchris** [comment]
Closing as this change is available on the `0.6.x-dev` version and will be release at `v0.6`
